ext/sodium/libsodium.c

```diff
@@ -1465,6 +1472,87 @@ PHP_FUNCTION(sodium_crypto_stream_xor)
 	RETURN_NEW_STR(ciphertext);
 }
 
+#ifdef crypto_stream_xchacha20_KEYBYTES
+PHP_FUNCTION(sodium_crypto_stream_xchacha20)
+{
+	zend_string   *ciphertext;
+	unsigned char *key;
+	unsigned char *nonce;
+	zend_long      ciphertext_len;
+	size_t         key_len;
+	size_t         nonce_len;
+
+	if (zend_parse_parameters(ZEND_NUM_ARGS(), "lss",
+									&ciphertext_len,
+									&nonce, &nonce_len,
+									&key, &key_len) == FAILURE) {
+		sodium_remove_param_values_from_backtrace(EG(exception));
+		RETURN_THROWS();
+	}
+	if (ciphertext_len <= 0 || ciphertext_len >= SIZE_MAX) {
+		zend_argument_error(sodium_exception_ce, 1, "length must be greater than 0");
+		RETURN_THROWS();
+	}
+	if (nonce_len != crypto_stream_xchacha20_NONCEBYTES) {
+		zend_argument_error(sodium_exception_ce, 2, "nonce must be SODIUM_CRYPTO_STREAM_XCHACHA20_NONCEBYTES bytes long");
+		RETURN_THROWS();
+	}
+	if (key_len != crypto_stream_xchacha20_KEYBYTES) {
+		zend_argument_error(sodium_exception_ce, 3, "key must be SODIUM_CRYPTO_STREAM_XCHACHA20_KEYBYTES bytes long");
+		RETURN_THROWS();
+	}
+	ciphertext = zend_string_checked_alloc((size_t) ciphertext_len, 0);
```

Comment:
Hm, it looks like this function was added in the PECL extension in https://github.com/jedisct1/libsodium-php/commit/75701f246f44911ca9cd559341c827538f563d24, but is missing here...

Reply:
`./configure` doesn't work locally, otherwise we would have caught these without having to depend on CI.